### PR TITLE
docs(connector): add zod suggestion and update connector contents

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. Add zod suggestion for creating an connector
2. Update connector contents due to changes in implementation

## Testing

N/A

### CODE REVIEW

File: docs/docs/recipes/create-your-connector/fragments/_index_contents.md [hunk 1]
```diff
@@ -26,7 +28,7 @@ interface SocialConnector extends BaseConnector {
 To see connector methods' definition and build a picture of connector interface design, see [`@logto/connector-types`](https://github.com/logto-io/logto/blob/master/packages/connector-types/src/index.ts). You can also find _ConnectorMetadata_ reference at "[Connectors](../../references/connectors)".
 
 - A connector `validateConfig` is obligatory for all connectors.
-- All _SMS Connectors_ and _Email Connectors_ require a `sendMessage` method to call providers message sending APIs.
+- All _SMS Connectors_ and _Email Connectors_ require a `sendMessage` method to call providers message sending APIs using configs from the database. Developers can choose to implement a `sendTestMessage` method to send a testing message with unsaved config in the text input box while setting connectors up.
```
Comment (demonzoo) [inline docs/docs/recipes/create-your-connector/fragments/_index_contents.md:31]:
```suggestion
- All _SMS Connectors_ and _Email Connectors_ require a `sendMessage` method to call providers message sending APIs using configs from the database. Developers can choose to implement a `sendTestMessage` method to send a testing message with unsaved config while setting connectors up.
```
The `text input box` makes me confused.

File: docs/docs/recipes/create-your-connector/fragments/_types_contents.md [hunk 2]
```diff
@@ -4,3 +4,9 @@ Typical values in _types.ts_:
 - Request parameter types
 - Response parameter types
 - Error types
+
+:::note
+We STRONGLY recommend using [_Zod_](https://github.com/colinhacks/zod) to build type guards for the connector's config, requests, and responses.
+
+Zod could be beneficial when trying to set up connectors, and the toast will show detailed error messages from ZodError if your config JSON is invalid.
```
Comment (demonzoo) [inline docs/docs/recipes/create-your-connector/fragments/_types_contents.md:11]:
```suggestion
Zod could be beneficial when trying to set up connectors, and a toast with detailed error messages from ZodError will show up if your config JSON is invalid.
```
Just being nit-picking

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/recipes/create-your-connector/fragments/_index_contents.md [hunk 3]
```diff
@@ -28,7 +30,7 @@ interface SocialConnector extends BaseConnector {
 关于 _ConnectorMetadata_ 的参考，见「[连接器](../../references/connectors)」。
 
 - 连接器的 `validateConfig` 方法对于所有类型的连接器都是必须的。
-- 所有 _邮件连接器_ 和 _短信连接器_ 都需要 `sendMessage` 方法来调用发送信息的 API。
+- 所有 _邮件连接器_ 和 _短信连接器_ 都需要 `sendMessage` 方法来调用发送信息的 API（使用的是数据库中的连接器配置）。开发者可以选择性地实现 `sendTestMessage` 方法来使用在设置连接器时尚未保存的配置发送测试信息。
```
Comment (demonzoo) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/recipes/create-your-connector/fragments/_index_contents.md:33]:
```suggestion
- 所有 _邮件连接器_ 和 _短信连接器_ 都需要 `sendMessage` 方法来调用发送信息的 API（使用的是数据库中的连接器配置）。开发者可以选择性地实现 `sendTestMessage` 方法，来在设置连接器时使用尚未保存的配置发送测试信息。
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/recipes/create-your-connector/fragments/_types_contents.md [hunk 4]
```diff
@@ -4,3 +4,9 @@ _types.ts_ 中的一些常见的变量：
 - 请求的参数类型定义
 - 响应的参数类型定义
 - 错误类型定义
+
+:::note
+我们 **强烈建议** 使用 [_Zod_](https://github.com/colinhacks/zod) 来建立对连接器 config、请求和相应的类型保护。
```
Comment (demonzoo) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/recipes/create-your-connector/fragments/_types_contents.md:9]:
```suggestion
我们 **强烈建议** 使用 [_Zod_](https://github.com/colinhacks/zod) 来建立对连接器配置、请求和响应的类型保护。
```

**Review by demonzoo — APPROVED** [review]

**Review by gao-sun — APPROVED** [review]

**Pull request merged**